src/blockrelay/graphene_set.h

```diff
@@ -131,18 +148,94 @@ class CGrapheneSet
     // of cheap hashes in the block in the correct order
     std::vector<uint64_t> Reconcile(const std::vector<uint256> &receiverItemHashes);
 
+    // This version assumes that we know the set that have passed through the sender Bloom filter
+    std::vector<uint64_t> Reconcile(const std::set<uint64_t> &setSenderFilterPositiveCheapHashes);
+
     // Pass a map of cheap hash to transaction hashes that the local machine has to reconcile with the remote and
     // return a list of cheap hashes in the block in the correct order
     std::vector<uint64_t> Reconcile(const std::map<uint64_t, uint256> &mapCheapHashes);
 
-    std::vector<uint64_t> Reconcile(std::set<uint64_t> &receiverSet, const CIblt &localIblt);
+    static std::vector<uint64_t> Reconcile(const std::map<uint64_t, uint256> &mapCheapHashes,
+        std::shared_ptr<CIblt> _pSetIblt,
+        std::shared_ptr<CBloomFilter> _pSetFilter,
+        std::shared_ptr<CVariableFastFilter> _pFastFilter,
+        std::vector<unsigned char> _encodedRank,
+        bool _computeOptimized,
+        bool _ordered);
+
+    static std::vector<uint64_t> Reconcile(const std::set<uint64_t> &setSenderFilterPositiveCheapHashes,
+        const CIblt &localIblt,
+        std::shared_ptr<CIblt> _pSetIblt,
+        std::vector<unsigned char> _encodedRank,
+        bool _ordered);
+
+    static GrapheneSetOptimizationParams DetermineGrapheneSetOptimizationParams(uint64_t nReceiverUniverseItems,
+        uint64_t nSenderUniverseItems,
+        uint64_t nItems,
+        uint64_t version);
+
+    static CIblt ConstructIblt(uint64_t nReceiverUniverseItems,
+        double optSymDiff,
+        double fBloomFPR,
+        uint32_t ibltSalt,
+        uint64_t graphenSetVersion,
+        uint64_t nOverrideValue);
 
     static std::vector<unsigned char> EncodeRank(std::vector<uint64_t> items, uint16_t nBitsPerItem);
 
     static std::vector<uint64_t> DecodeRank(std::vector<unsigned char> encoded, size_t nItems, uint16_t nBitsPerItem);
 
     static double BloomFalsePositiveRate(double optSymDiff, uint64_t nReceiverExcessItems);
 
+    static double TruePositiveMargin(uint64_t nSenderFilterPositiveItems,
+        uint64_t nReceiverUniverseItems,
+        double fSenderBloomFpr,
+        uint64_t nLowerBoundTruePositives);
```

Comment:
Just took a very quick pass through the code...quite a lot to digest but the only obvious superficial item is the naming convention used for the variables...particularly the prefix you're using for doubles. The "f" you'll find throughout the codebase is used for indicating a bool (flag) and  not for floating points...doubles are sometimes seen with a "d" prefix or an "n". Seems that sometimes you use a prefix then sometimes not.  I think it's good practice, at the least, to be consistent.  Some devs disagree but  I find it does help for the reviewer to have these conventions, otherwise, it's tougher for some reviewers (like me) and with very large PR's it just makes it more likely you'll get a good review of the code (Just my two cents)..other than that it looks like you've done a lot of work and I'm looking forward to sifting though it and doing some testing.

Reply:
Sure, I'm happy to make that change. Which prefix do you prefer for doubles?

Reply:
Based on our offline discussion, I decided to just removed the "f" in front of the new doubles created in this PR.